src/item-detail/item-detail.js

```diff
@@ -35,18 +35,37 @@ class ItemDetail extends Component {
     constructor(props) {
         super(props);
 
-        this.state = {...initState};
+        this.state = {
+            ...initState,
+            description: props.params.description
+        };
 
         this.onChangeTypeOfContent = this.onChangeTypeOfContent.bind(this);
+        this.onUpdateDescription = this.onUpdateDescription.bind(this);
+        this.onSaveDescription = this.onSaveDescription.bind(this);
     }
 
     onChangeTypeOfContent(isEditable) {
         this.setState({isEditable})
     }
 
+    onUpdateDescription(newDescription) {
+        this.setState({
+            description: newDescription
+        })
+    }
+
+    onSaveDescription(){
+        this.props.onChangeDescription(this.state.description);
+        this.setState({
+            isEditable: false
+        });
```

Comment:
`onUpdateDescription` was implemented for it.

Reply:
`onUpdateDescription` takes care of updating the data of the `ItemDescription` component, not to saving it to the App state. It runs when the description changes to change the value of what's displayed in `ItemDetails`.

This function `onSaveDescription` runs `this.props.onChangeDescription` which should only run when the description is *saved*, not *changed*. This changes the description in App.